src/sidebar/components/NotebookView.js

```diff
@@ -21,37 +23,88 @@ import ThreadList from './ThreadList';
  */
 function NotebookView({ loadAnnotationsService }) {
   const store = useStoreProxy();
+
+  const filters = store.getFilterValues();
   const focusedGroup = store.focusedGroup();
+  const forcedVisibleCount = store.forcedVisibleThreads().length;
+  const hasAppliedFilter = store.hasAppliedFilter();
+  const isLoading = store.isLoading();
+  const resultCount = store.annotationResultCount();
+
+  const rootThread = useRootThread();
 
-  // Reload annotations if focused group changes
+  const groupName = focusedGroup?.name ?? '…';
+
+  const lastPaginationPage = useRef(1);
+  const [paginationPage, setPaginationPage] = useState(1);
+
+  // Load all annotations; re-load if `focusedGroup` changes
   useEffect(() => {
     // Load all annotations in the group, unless there are more than 5000
-    // of them: this is a performance safety valve
+    // of them: this is a performance safety valve.
+
+    // NB: In current implementation, this will only happen/load once (initial
+    // annotation fetch on application startup), as there is no mechanism
+    // within the Notebook to change the `focusedGroup`. If the focused group
+    // is changed within the sidebar and the Notebook re-opened, an entirely
+    // new iFrame/app is created. This will need to be revisited.
     store.setSortKey('Newest');
     if (focusedGroup) {
       loadAnnotationsService.load({
         groupId: focusedGroup.id,
         maxResults: 5000,
       });
+      setPaginationPage(1);
```

Comment:
The idea is that we'd want to reset pagination after reloading new results. In real life right now, this block only executes once at app startup (see notes in comments above), but in the future, that will likely change.